File: mslibrary-kotlindocs/src/main/java/kr/msleague/message/adapter/ItemStackAdapter.java
package kr.msleague.message.adapter;

import kr.msleague.message.api.MSMessageAdapter;
import org.bukkit.inventory.ItemStack;

/**
 * MSLibrary 에서 기본적으로 제공하는 ItemStack 어댑터입니다.
 */
public class ItemStackAdapter implements MSMessageAdapter<ItemStack> {
    @Override
    public String reformat(String origin, ItemStack obj) {
        return "";
    }
}

File: mslibrary-message/src/main/java/kr/msleague/message/adapter/ItemStackAdapter.kt
package kr.msleague.message.adapter

import kr.msleague.message.api.MSMessageAdapter
import kr.msleague.util.locale.friendlyName
import kr.msleague.util.locale.l10nDisplayName
import org.bukkit.inventory.ItemStack

class ItemStackAdapter : MSMessageAdapter<ItemStack> {

    override fun request(obj: ItemStack, placeHolder: String): String? {
        return when (placeHolder) {
            "item_name" -> obj.l10nDisplayName
            "item_amount" -> obj.amount.toString()
            "item_type" -> obj.friendlyName
            else -> null
        }
    }
}